Libplanet/Tx/AddressSet.cs

```diff
@@ -0,0 +1,212 @@
+            new AddressSet(_addresses.Intersect(other).ToImmutableArray(), bypassCheck: true);
+
+        /// <inheritdoc cref="IImmutableSet{T}.Except(IEnumerable{T})"/>
+        [Pure]
+        public IImmutableSet<Address> Except(IEnumerable<Address> other) =>
+            new AddressSet(_addresses.Except(other).ToImmutableArray(), bypassCheck: true);
+
+        /// <inheritdoc cref="IImmutableSet{T}.SymmetricExcept(IEnumerable{T})"/>
+        [Pure]
+        public IImmutableSet<Address> SymmetricExcept(IEnumerable<Address> other)
+        {
+            // TODO: test
+            ICollection<Address> operand = other is ICollection<Address> o
+                ? o
+                : other.ToArray();
+            return new AddressSet(_addresses.Union(operand).Except(_addresses.Intersect(operand)));
+        }
+
+        /// <inheritdoc cref="IImmutableSet{T}.Union(IEnumerable{T})"/>
+        [Pure]
+        public IImmutableSet<Address> Union(IEnumerable<Address> other) =>
+            new AddressSet(_addresses.Union(other));
+
+        /// <inheritdoc cref="IImmutableSet{T}.SetEquals(IEnumerable{T})"/>
+        [Pure]
+        public bool SetEquals(IEnumerable<Address> other)
+        {
+            // TODO: test
+            ICollection<Address> operand = other is ICollection<Address> o
+                ? o
+                : other.ToArray();
+            if (_addresses.Length != operand.Count)
+            {
+                return false;
+            }
+
+            return _addresses.All(operand.Contains);
+        }
+
+        /// <inheritdoc cref="IImmutableSet{T}.IsProperSubsetOf(IEnumerable{T})"/>
+        [Pure]
+        public bool IsProperSubsetOf(IEnumerable<Address> other)
+        {
+            // TODO: test
+            ISet<Address> operand = other is ISet<Address> o
+                ? o
+                : other.ToImmutableHashSet();
+            return operand.IsProperSupersetOf(_addresses);
+        }
+
+        /// <inheritdoc cref="IImmutableSet{T}.IsProperSupersetOf(IEnumerable{T})"/>
+        [Pure]
+        public bool IsProperSupersetOf(IEnumerable<Address> other)
+        {
+            // TODO: test
+            ISet<Address> operand = other is ISet<Address> o
+                ? o
+                : other.ToImmutableHashSet();
+            return operand.IsProperSubsetOf(_addresses);
+        }
+
+        /// <inheritdoc cref="IImmutableSet{T}.IsSubsetOf(IEnumerable{T})"/>
+        [Pure]
+        public bool IsSubsetOf(IEnumerable<Address> other)
+        {
+            // TODO: test
+            ISet<Address> operand = other is ISet<Address> o
+                ? o
+                : other.ToImmutableHashSet();
+            return operand.IsSupersetOf(_addresses);
+        }
+
+        /// <inheritdoc cref="IImmutableSet{T}.IsSupersetOf(IEnumerable{T})"/>
+        [Pure]
+        public bool IsSupersetOf(IEnumerable<Address> other)
+        {
+            // TODO: test
```

Comment:
I think you forgot to remove the TODO: test commands. 😋

Reply:
Amended!